src/server/src/main/java/io/cassandrareaper/ReaperApplication.java

```diff
@@ -232,13 +232,16 @@ public void run(ReaperApplicationConfiguration config, Environment environment)
     environment.jersey().register(addCryptoResource);
 
     HttpClient httpClient = createHttpClient(config, environment);
-    ScheduledExecutorService ses = environment.lifecycle().scheduledExecutorService("Diagnostics").threads(6).build();
-    final DiagEventSubscriptionResource eventsResource = new DiagEventSubscriptionResource(context, httpClient, ses,
-        context.storage.getEventsDao());
-    environment.jersey().register(eventsResource);
-    final DiagEventSseResource diagEvents = new DiagEventSseResource(context, httpClient, ses,
-        context.storage.getEventsDao());
-    environment.jersey().register(diagEvents);
+
+    if (context.managementConnectionFactory instanceof JmxManagementConnectionFactory) {
```

Comment:
I'm not entirely sure where the hate for instanceof comes from, but from the same file, we can of course this method instead of instanceof:

```java
if (context.config.getHttpManagement() == null || !context.config.getHttpManagement().getEnabled()) {
```

Because that's what the initialization of HTTP or JMX endpoint uses. Not sure how that's different to instanceof check, both are single lines and have the exact same semantics in this case. Especially since it's done in the same file. Actually, I'd even say that instance of makes more sense since it's a single check and not dual one (and we don't repeat the same procedure everywhere then). 

So why pollute interface with the info, when the other parts don't use such either?

Reply:
That's fine. I think what I would like to see is a more involved refactor later on. The hate for `instanceof` isn't hate on it so much (it's in the JDK), it's just that it indicates the interface design is sub-optimal here. Ideally we never have to use it. But to do so would require redesigning a lot of Reaper.

I'm fine with the code PR as it is.

Reply:
This conversation was EXACTLY my thought process too :) I also prefer using the config since config is globally available and immutable, and doesn't break the design of the types. But in the end the result is the same, so the bigger refactor is probably the most important thing in the end.

Reply:
I'm infinitely interested in how this upcoming refactor removes the initializeManagement() method and the new instance creation of JMX/HTTPManagementProxy. 

https://github.com/thelastpickle/cassandra-reaper/blob/master/src/server/src/main/java/io/cassandrareaper/ReaperApplication.java#L304

Because I'm super puzzled how we create that instance in that case. And what on earth is the benefit of avoiding all this.

Reply:
Let's talk about the upcoming refactor when we do it, and stay focused on getting the current milestone's work out for now.